### PR TITLE
JUCX: compare native structs by id.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Implement equals and hashCode for native structs.

## Why ?
To compare endpoints that may be created in different places to make sure it's one underlying ucx endpoint.

### CODE REVIEW

File: bindings/java/src/test/java/org/openucx/jucx/UcpEndpointTest.java [hunk 1]
```diff
@@ -655,6 +664,12 @@ public void testActiveMessages() throws Exception {
             } catch (Exception e) {
                 e.printStackTrace();
             }
+
+            if (cachedEp.get() != null) {
+                assertEquals(cachedEp.get(), replyEp);
+            } else {
+                cachedEp.set(replyEp);
+            }
```
Comment (yosefe) [inline bindings/java/src/test/java/org/openucx/jucx/UcpEndpointTest.java:672]: test also hashCode?

**Review by yosefe — APPROVED** [review]

**Comment by petro-rudenko** [comment]
There's an issue with NativeId. HashCode should be taken out of the immutable field.  Otherwise, if an object is in hashMap and we close it - we set NativeId to null.

**Comment by petro-rudenko** [comment]
I'll check what we can do with it, like to create another field nativeId_hash that won't be set to null

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by petro-rudenko** [comment]
bot:pipe:retest

**Review by yosefe — APPROVED** [review]

**Pull request merged**